scrypto/src/resource/non_fungible_id.rs

```diff
@@ -65,18 +63,38 @@ impl fmt::Display for ParseNonFungibleIdError {
 // binary
 //========
 
+fn validate_id(slice: &[u8]) -> Result<(), DecodeError> {
+    let mut decoder = ScryptoDecoder::new(slice);
```

Comment:
I think it'd be worth adding a comment explaining why we're manually doing this instead of using an any encoder, as someone naive might just add it back in.